x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/ingest/InferenceProcessor.java

```diff
@@ -272,5 +288,10 @@ void checkSupportedVersion(InferenceConfig config) {
                     minNodeVersion));
             }
         }
+
+        @Override
+        public void licenseStateChanged() {
+            this.inferenceAllowed = licenseState.isMachineLearningAllowed();
```

Comment:
What is the mechanism which updates `licenseState`? 
Am I right that it is a mutable object and the license listener notifies `licenseStateChanged` method after the internal state of `licenseState` object changes?

Reply:
Yes, it is a singleton that gets mutated in place. See other examples in `org.elasticsearch.xpack.ml.InvalidLicenseEnforcer`